src/adapter/src/coord.rs

```diff
@@ -2679,6 +2697,9 @@ impl Coordinator {
             // no need to install a dataflow in the first place.
             if write_frontier.is_empty() {
                 Antichain::new()
+            } else if self.controller.read_only() {
+                // See note above, where we pick index as_ofs.
+                min_as_of.clone()
```

Comment:
I think we can just have this special case once below in `let as_of = ...`, rather than repeating it in the index/MV-specific branches?

Reply:
Third time's the charm, I pushed that change 👌